### ISSUE TITLE
Sync doesn't keep local folder structure

### ISSUE THREAD

**Issue by marcelreppi** [moritzgloeckl/overleaf-sync#20]
Hey :)

When I sync my local project, which contains multiple subfolders with `.tex` files, all files are added to the root directory of the Overleaf project and no subfolders are created.

Is there an option that I am missing?

Thanks for your efforts!

**Comment by moritzgloeckl** [comment]
Hi Marcel,

I just tried syncing with multiple subfolders and everything seems to be working fine on my end. Are you sure you're on the latest version?

If you run:

`ols --version`

does it say:

`ols, version 1.1.2`

The latest version is `1.1.2`. Previous versions do not support subfolders.

Hope this helps.

**Comment by marcelreppi** [comment]
Hey Moritz,

thanks for your response!
I checked and I am indeed on version `1.1.2`.

I am mainly using the `-l` option to keep my Overleaf project in sync with my local files.

I tested it again with a blank Overleaf project and I still get no subfolders.
Here are some screenshots.
If there is anything more I can do to help find the issue please let me know :)

Local file structure:
![files-local](https://user-images.githubusercontent.com/22420478/104214064-4c00fe00-5437-11eb-8a28-c825d665792c.png)


Logs of executing `ols -l`:
![log](https://user-images.githubusercontent.com/22420478/104214076-4f948500-5437-11eb-99b5-f3a5971a288b.png)


Resulting Overleaf folder structure:
![files-ol](https://user-images.githubusercontent.com/22420478/104214090-54593900-5437-11eb-881c-f19c735fa27f.png)

**Comment by moritzgloeckl** [comment]
Hi there,

I just tested this again with the same file names and also using the `-l` option. I still can't reproduce this unfortunately. Can you please share your `.olignore` file too? 

Also is there anything specific about your Overleaf account? Are you on a Premium plan?

Thanks

**Comment by marcelreppi** [comment]
The `.olignore` file contains this:

```
build/**

*.zip

*log

.olauth
```

I have also tried it when I delete the `.olignore` file and it still doesn't work.

I am on the regular Free Plan and my Overleaf account is linked to my Google Account.

Thanks for your efforts!

**Comment by marcelreppi** [comment]
Okay I am using Windows with Git Bash als my main console.
I have just tried installing overleaf-sync on my WSL and try to sync my project using the WSL and it works correctly!
It seems that it doesn't work correctly when using the regular Windows CMD nor the Git Bash on Windows but using the Ubuntu WSL CLI it works correctly.

**Comment by marcelreppi** [comment]
But even with the WSL I get some quirky behavior although it is an empty project on Overlead... some files don't get uploaded at all and some subfolders (e.g. `cv` and `coverLetter`) still end up in the root directory.
Maybe you need some artificial delay in uploading files to Overleaf so that the server has enough time to process all requests?

![Group 6](https://user-images.githubusercontent.com/22420478/104456935-b5a11980-55a9-11eb-8c7f-b897723239cb.png)
![Group 5](https://user-images.githubusercontent.com/22420478/104456944-b8037380-55a9-11eb-8250-b126472ad445.png)

**Comment by moritzgloeckl** [comment]
Thank you for taking a look into this! It seems there's maybe something going wrong with your Windows shell? I'm on a Unix system so this is why I might've not been able to reproduce. But it should work regardless since I'm using standard Python OS functions. 

Furthermore, I'm going to test huge repositories like yours again to check if there are any problems with uploading.

**Comment by PrashanthaTP** [comment]
Hey ,😃,
I am too facing the same issue
I am using windows 10 , python 3.7.2


----

### Probable cause for the issue

Since the logic for  detecting whether folders need to be created uses hard coded path seperator ("/" here),and in windows the
path seperator is "\\" so the create_folder function will not be called.

https://github.com/moritzgloeckl/overleaf-sync/blob/4b77df9e04a762e2e14ad8812322e6c7663fa296/olsync/olclient.py#L207-L222

### solution ?
using standard module **os**
changing 
https://github.com/moritzgloeckl/overleaf-sync/blob/4b77df9e04a762e2e14ad8812322e6c7663fa296/olsync/olclient.py#L207-L208
to 
```python
        if os.sep in file_name:
            local_folders = file_name.split(os.sep)[:-1]  # Remove last item since this is the file name
```
worked for me 

### problems faced
When I had more than one subfolder
like 

folder
|--subfolder_1
|--subfolder_2

when for first time I run the ols command only subfolder_1 created and ends in [ERROR] An error occurred while creating new file(s) on [remote]
second time sub_folder_2 gets created

---
btw , Thank you for creating this library👍

**Comment by moritzgloeckl** [comment]
Should be fixed with #40 now! Thank you @seapat and @PrashanthaTP

**Comment by ValeryVerkhoturov** [comment]
Issue was not fixed.
```
Syncing files from [local] to [remote]
========================================

[NEW] Following new file(s) created on [remote]
        Images\image1.jpg
        Images\image2.jpg
        Images\tableeditor.png
        Images\tutor1.png
        Images\tutor2.png
        Listings\Main.java
        Listings\main.py
        Settings\FiraCode-Regular.otf
        Settings\format.tex
        Settings\listings.tex
        Settings\packages.tex
        TitlePages\tasklist.tex
        TitlePages\test_examination_form.tex

[UPDATE] Following file(s) updated on [remote]

[SYNC] Following file(s) are up to date
        bibpage.tex
        content.tex
        macros.tex
        main.tex
        README.md
        refs.bib
        titlepage.pdf
        tocpage.tex
        tutorial.tex

[SKIP] Following file(s) on [local] have not been synced to [remote]

✅  Synced files from [local] to [remote]

C:\Users\User\latex>ols --version
ols, version 1.1.6

C:\Users\User\latex>python
Python 3.10.2 (tags/v3.10.2:a58ebcc, Jan 17 2022, 14:12:15) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> os.sep
'\\'
```
